src/components/Swap/SwapActionButton/SwitchChainButton.tsx

```diff
@@ -1,10 +1,42 @@
 import { Trans } from '@lingui/macro'
+import { useWeb3React } from '@web3-react/core'
 import ActionButton from 'components/ActionButton'
 import useSwitchChain from 'hooks/useSwitchChain'
 import { Spinner } from 'icons'
-import { useCallback, useMemo, useState } from 'react'
+import { useCallback, useEffect, useMemo, useState } from 'react'
 import { Colors } from 'theme'
 
+interface ChainIds {
+  inputChainId?: number
+  outputChainId?: number
+}
+
+function useDesiredChainId({ inputChainId, outputChainId }: ChainIds): number | undefined {
+  const { chainId } = useWeb3React()
+  if (chainId === undefined) return
+
+  const desiredChainId = inputChainId ?? outputChainId
+  if (desiredChainId === chainId) return
+
+  return desiredChainId
+}
```

Comment:
It's not clear from the implementation when I'd want to use this function, but it reads like a general purpose utility. Can you add a doc comment at the top explaining why it exists?